src/pgwire/src/protocol.rs

```diff
@@ -407,7 +428,12 @@ where
     }
 
     async fn advance_drain(&mut self) -> Result<State, io::Error> {
-        match self.conn.recv().await? {
+        let message = self.conn.recv().await?;
+        if message.is_some() {
```

Comment:
Should this get moved inside of `fn sync()`? I don't think the `Drain` state is the correct place to disable the timeout. `Drain` only seems to happen when an error occurs (it's only produced in `error()` and `aborted_txn_error()`).

Reply:
If the user is in an aborted transaction, and has not sent a `COMMIT`, `ROLLBACK`, or `Sync`, then the state will be `Drain`. If the user continues to send a non-`Sync` message, the timeout timer should be restarted, which is why it needs to be outside of `sync`. There's a specific test for this in `sql`.

For example if someone does the following
```
joe=# BEGIN;
BEGIN
joe=*# SELECT 1/0;
ERROR:  division by zero
joe=!# SELECT 1;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
joe=!# SELECT 1;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
joe=!# SELECT 1;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
joe=!# SELECT 1;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
joe=!# SELECT 1;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
...
```
then each `SELECT` should restart the timer.